File: api/src/main/java/kimosabe/api/model/Role.java
package kimosabe.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;

@Table(name = "roles")
@Entity
@NoArgsConstructor @Getter @Setter
public class Role implements Serializable {
    @Id @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Enumerated(EnumType.STRING)
    private RoleName name;

    public Role(RoleName name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object obj) {
        if(obj instanceof Role){
            Role toCompare = (Role) obj;
            return this.name.equals(toCompare.name);
        }
        return false;
    }
    @Override
    public int hashCode() {
        return name.hashCode();
    }
}

File: api/src/main/kotlin/com/kimosabe/api/model/Role.kt
package com.kimosabe.api.model

import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name="roles")
class Role(
        @Id @GeneratedValue(strategy = GenerationType.SEQUENCE) var id : Long? = null,
        @Enumerated(EnumType.STRING) var name: RoleName? = null
) : Serializable